src/amplitude-client.js

```diff
@@ -1908,11 +1911,9 @@ AmplitudeClient.prototype.__VERSION__ = function getVersion() {
  * @param {string} name - Custom library name
  * @param {string} version - Custom library version
  */
-AmplitudeClient.prototype.setLibrary = function setLibrary(
-  name = this.options.libraryName,
-  version = this.options.libraryVersion,
-) {
-  this.options.library = { name: name, version: version };
+AmplitudeClient.prototype.setLibrary = function setLibrary(name, version) {
+  if (name) this.options.library.name = name;
+  if (version) this.options.library.version = version;
```

Comment:
I think we should do an explicit check for `null` or `undefined`. Edge case conditions for falsey values may lead to unexpected outcomes (ike `0` or `''` )